service/history/outbound_queue_factory.go

```diff
@@ -75,6 +93,15 @@ func NewOutboundQueueFactory(params outboundQueueFactoryParams) QueueFactory {
 		return quotas.NewDefaultOutgoingRateLimiter(func() float64 { return 100.0 })
 	})
 	grouper := queues.GrouperStateMachineNamespaceIDAndDestination{}
+	circuitBreaker := gobreaker.NewCircuitBreaker(gobreaker.Settings{
```

Comment:
You're going to want a circuit breaker per `StateMachineTaskTypeNamespaceIDAndDestination` initially, later on we may decide to use a per `StateMachineTaskTypeAndDestination` breaker.